api/tests/functional-tests/backend/metrics/test_text_generation.py

```diff
@@ -348,48 +375,397 @@ def test_text_generation(
+            "rouge1": 0.67,
+            "rouge2": 0.5,
+            "rougeL": 0.67,
+            "rougeLsum": 0.67,
+        },  # off by one
+        {
+            "prediction": "flipping the roaring white dolphin",
+            "references": ["flip the roaring white dolphin"],
+            "rouge1": 0.8,
+            "rouge2": 0.75,
+            "rougeL": 0.8,
+            "rougeLsum": 0.8,
+            "use_stemmer": False,
+        },  # incorrect match without stemming
+        {
+            "prediction": "flipping the roaring white dolphin",
+            "references": ["flip the roaring white dolphin"],
+            "rouge1": 1,
+            "rouge2": 1,
+            "rougeL": 1,
+            "rougeLsum": 1,
+            "use_stemmer": True,
+        },  # correct match with stemming
+        {
+            "prediction": "flipping the roaring white dolphin",
+            "references": [
+                "some random sentence",
+                "some other sentence",
+                "some final reference",
+                "flip the roaring white dolphin",
+            ],
+            "rouge1": 1,
+            "rouge2": 1,
+            "rougeL": 1,
+            "rougeLsum": 1,
+            "use_stemmer": True,
+        },  # test multiple references
+    ]
+
+    multiple_prediction_examples = [
+        {
+            "prediction": ["Mary loves Joe", "Mary loves Jack"],
+            "references": [
+                ["Mary loves June", "some other sentence"],
+                ["some other sentence", "the big fox hunts rabbits"],
+            ],
+            "expected_value": [
+                {
+                    "prediction": "Mary loves Joe",
+                    "value": {
+                        "rouge1": 0.6666666666666666,
+                        "rouge2": 0.5,
+                        "rougeL": 0.6666666666666666,
+                        "rougeLsum": 0.6666666666666666,
+                    },
+                },
+                {
+                    "prediction": "Mary loves Jack",
+                    "value": {
+                        "rouge1": 0.0,
+                        "rouge2": 0.0,
+                        "rougeL": 0.0,
+                        "rougeLsum": 0.0,
+                    },
+                },
+            ],
+        },  # off by one
+        {
+            "prediction": [
+                "flipping the roaring white dolphin",
+                "Mary loves Joe",
+            ],
+            "references": [
+                [
+                    "some random sentence",
+                    "some other sentence",
+                    "some final reference",
+                    "flip the roaring white dolphin",
+                ],
+                ["beep bop", "Mary loves June"],
+            ],
+            "expected_value": [
+                {
+                    "prediction": "flipping the roaring white dolphin",
+                    "value": {
+                        "rouge1": 1.0,
```

Comment:
we're setting `"use_stemmer"`:True here, so "flipping" becomes "flip" (since the suffix gets trimmed). that means we should expect a score of 1 since that was the only difference between prediction and reference